src/EditProvider.ts

```diff
@@ -99,16 +100,84 @@ export class EditProvider
   }
 
   public static beautifyOptions(
-    filePath: string = vscode.workspace.rootPath
+    path: string = vscode.workspace.rootPath
   ): Promise<LanguageOptionValues> {
     try {
-      const explorer = cosmiconfig("unibeautify", {});
+      /**
+       * Use a different `.unibeautify` config when vscode config is set and
+       * define if the project config should extend the vscode config.
+       *
+       * @argument {unibeautify.defaultConfig} string path to a file
+       * @argument {unibeautify.extendConfig} boolean extend or not
+       */
+      const vscodeSettings = vscode.workspace.getConfiguration("unibeautify");
+      let defaultConfigOverride = vscodeSettings.defaultConfig;
+      let extendConfig = vscodeSettings.extendConfig;
+      let workspace = path;
+
+      const explorerOptions: any = {};
+      if (defaultConfigOverride) {
+        path = defaultConfigOverride;
```

Comment:
The variable `path` is passed as an argument into this function. It is bad to reassign its value.

See these linting rules:
- https://eslint.org/docs/rules/no-param-reassign
- https://palantir.github.io/tslint/rules/no-parameter-reassignment/

We should add the linting rule to our Travis CI checks, if it is not already. cc @szeck87